### DISCUSSION TITLE
What is the idea behind Theia in the Cloud?

### DISCUSSION THREAD

**Discussion by jeanlucburot** [eclipse-theia/theia#9845] [Q&A]
It is mentioned in some documents that Theia can easily run within a cloud. Now, there are many ways I can think of running something in a cloud, as basically a cloud ist just a collection of computers sharing load. Specifically, the first thing I would think of is creating a Docker container, putting Theia and all that Theia needs into it and running that in a cloud. While certainly that would be possible and IMHO relatively easy to achieve, I wonder if that it what the Theia developers think about when putting Theia into the cloud. Also, I wonder about what cloud exactly as there are quite a few options starting with AWS, Azure and Google Cloud.

What is the idea behind putting Theia in the cloud?
Which cloud are we speaking of (if any specifically)?

**Comment by akosyakov** [comment]
@jeanlucburot Generally, Theia is designed to be agnostic to deployment target, so it can be deployed anywhere you want as a usual web application and does not require Docker. But yes, it is a typical case to put it in Docker and deploy with some cloud provider or self-host it. I hope you've seen already https://github.com/theia-ide/theia-apps for Docker image examples.

**Comment by jeanlucburot** [comment]
@akosyakov So I checked all the examples and their Dockerfile files. Thank you for the link. Unfortunately, I am still missing a piece of information.

How many resources (RAM, CPU, etc.) does Theia (basically Theia run in Node) need? As an example I have the simplest-most setup in mind like [this one](https://github.com/theia-ide/theia-apps/tree/master/theia-docker) or the same setup, but with Ubuntu. The reason I ask this is to know how many instances of Theia I am able to run on one single machine. Based on that, I would need to evaluate how much resources I provide to one instance when run in a Docker Swarm Mode, i.e. provide senseful entry values into the YAML file for Docker Compose, like:
```
    deploy:
      replicas: 5
      restart_policy:
        condition: on-failure
      resources:
        limits:
          cpus: "0.1"
          memory: 50M
```
